bindings.go

```diff
@@ -316,12 +317,15 @@ func AddPiece(
 	resPtr := C.sector_builder_ffi_add_piece(
 		(*C.sector_builder_ffi_SectorBuilder)(sectorBuilderPtr),
 		cPieceKey,
-		C.uint64_t(pieceSize),
-		cPiecePath,
+		C.uintptr(pieceFd),
+		C.uint64_t(pieceBytes),
 		C.uint64_t(pieceExpiryUtcSeconds),
 	)
 	defer C.sector_builder_ffi_destroy_add_piece_response(resPtr)
 
+	// Make sure our filedescriptor stays alive, stayin alive
+	runtime.KeepAlive(pieceFd)
```

Comment:
This needs to be the `pieceFile os.File` not the `pieceFd`. Otherwise, you are keeping the uint alive instead of the `os.File`.

Reply:
fixed